packages/extension/src/shared/transactionReview.service.ts

```diff
@@ -212,3 +220,22 @@ export const getTransactionReviewHasSwap = (
 ) => {
   return !!getTransactionReviewSwap(transactionReview)
 }
+
+export const getTransactionReviewWithType = (
+  transactionReview?: ApiTransactionReviewResponse,
+): TransactionReviewWithType | undefined => {
+  if (!transactionReview) {
+    return
+  }
+  for (const review of transactionReview.reviews) {
+    if (
+      review.activity?.type &&
+      apiTransactionReviewActivityType[review.activity?.type]
```

Comment:
`review.activity?.type` is already of type `ApiTransactionReviewActivityType`. Why do you double check this?

If still needed this could look like
```suggestion
      apiTransactionReviewActivityType.includes(review.activity.type)
```